include/CMakeLists.txt

```diff
@@ -38,3 +38,10 @@ set(INCLUDE_SOURCE_FILES
 
 add_library(${PROJECT_NAME} INTERFACE ${INCLUDE_SOURCE_FILES})
 target_include_directories(${PROJECT_NAME} INTERFACE ${CMAKE_CURRENT_SOURCE_DIR})
+
+set(INSTALL_DESTINATION ${CMAKE_INSTALL_PREFIX}/libfn)
+
+install(FILES ${INCLUDE_SOURCE_FILES} 
+        DESTINATION ${INSTALL_DESTINATION}/functional)
```

Comment:
... this destination should be `${INSTALL_DESTINATION}/include/functional` and also ...

Reply:
Actually this should be a relative path. CMake makes it implicitly relative to either of `$CMAKE_INSTALL_PREFIX` or `--prefix` parameter of the `install` target